frontend/src/middleware.ts

```diff
@@ -0,0 +1,44 @@
+import { NextRequest, NextResponse } from "next/server";
+
+export const config = {
+  matcher: "/:path*",
+};
+
+export async function middleware(request: NextRequest) {
+  const baseUrl = "http://localhost:3000";
```

Comment:
for frontend especially, I feel like we should be more mindful of the use of the urls and ports from dev, so that we can have less issues when merging

```
if process.env.NODE_ENV == production:
// use production url
else:
// construct url from env variables
```

Reply:
Good idea! Thank you :D